dags/dts_inspector_priority.py

```diff
@@ -0,0 +1,163 @@
+
+DEFAULT_ARGS = {
+    "owner": "airflow",
+    "depends_on_past": False,
+    "start_date": datetime(2015, 1, 1, tz="America/Chicago"),
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 0,
+    "execution_timeout": duration(minutes=30),
+    "on_failure_callback": task_fail_slack_alert,
+}
+
+REQUIRED_SECRETS = {
+    # Socrata
+    "SO_KEY": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.apiKeyId",
+    },
+    "SO_SECRET": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.apiKeySecret",
+    },
+    "SO_TOKEN": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.appToken",
+    },
+    "SO_WEB": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.endpoint",
+    },
+    "PRIORITY_DATASET": {
+        "opitem": "dts-right-of-way-reporting",
+        "opfield": "datasets.Priority",
+    },
+    "SEGMENT_DATASET": {
+        "opitem": "dts-right-of-way-reporting",
+        "opfield": "datasets.Segments",
+    },
+    # AMANDA
+    "HOST": {
+        "opitem": "Amanda Read-Only (RO) replica database",
+        "opfield": "production.host",
+    },
+    "PORT": {
+        "opitem": "Amanda Read-Only (RO) replica database",
+        "opfield": "production.port",
+    },
+    "SERVICE_NAME": {
+        "opitem": "Amanda Read-Only (RO) replica database",
+        "opfield": "production.service",
+    },
+    "DB_USER": {
+        "opitem": "Amanda Read-Only (RO) replica database",
+        "opfield": "production.username",
+    },
+    "DB_PASS": {
+        "opitem": "Amanda Read-Only (RO) replica database",
+        "opfield": "production.password",
+    },
+    # S3
+    "BUCKET_NAME": {
+        "opitem": "atd-executive-dashboard",
+        "opfield": "production.Bucket",
+    },
+    "EXEC_DASH_PASS": {
+        "opitem": "atd-executive-dashboard",
+        "opfield": "production.AWS Secret Access Key",
+    },
+    "EXEC_DASH_ACCESS_ID": {
+        "opitem": "atd-executive-dashboard",
+        "opfield": "production.AWS Access ID",
+    },
+    # ArcGIS Online
+    "AGOL_USERNAME": {
+        "opitem": "AGOL Scripts Publisher",
+        "opfield": "production.Username",
+    },
+    "AGOL_PASSWORD": {
+        "opitem": "AGOL Scripts Publisher",
+        "opfield": "production.Password",
+    },
+}
+
+with DAG(
+    dag_id="dts_inspector_priority",
+    description="Downloads permits and road segment data from AMANDA and scores permits based on several metrics",
+    default_args=DEFAULT_ARGS,
+    schedule_interval="0 3 * * *" if DEPLOYMENT_ENVIRONMENT == "production" else None,
+    tags=["repo:dts-right-of-way-reporting", "amanda", "socrata", "permits"],
+    catchup=False,
+) as dag:
+    docker_image = "atddocker/dts-right-of-way-reporting:production"
+
+    env_vars = get_env_vars_task(REQUIRED_SECRETS)
+
+    t1 = DockerOperator(
```

Comment:
Not requesting a change here, per se, but I do want to share a few thoughts with the larger group:

I don't think this is wrong at all, but I think it's a little bit of a foot gun that we rely on an implicit task ordering mechanism to get `env_vars` processed before any of the following tasks, and then use the airflow v1 way of explicitly ordering tasks as seen below in the `t1 >> t2 >> t3 >> t4` line. 

I also have never understood why we don't use meaningful names there, but I recognize that it's a habit taken from old airflow docs. I guess i never understood why they didn't use meaningful names. Maybe it was just an example with dummy tasks.